crates/sncast/tests/e2e/script/general.rs

```diff
@@ -318,3 +322,177 @@ async fn test_missing_field() {
+        "--accounts-file",
+        accounts_json_path.as_str(),
+        "--account",
+        "user4",
+        "--url",
+        URL,
+        "script",
+        "run",
+        &script_name,
+    ];
+
+    let snapbox = runner(&args).current_dir(script_dir.path());
+
+    snapbox.assert().success().stdout_matches(indoc! {r"
+        ...
+        command: script run
+        status: success
+    "});
+
+    let state_file_path = Utf8PathBuf::from_path_buf(
+        script_dir
+            .path()
+            .join(get_default_state_file_name(script_name, "alpha-goerli")),
+    )
+    .unwrap();
+    let tx_entries_after_first_run = read_txs_from_state_file(&state_file_path).unwrap().unwrap();
+
+    assert_eq!(tx_entries_after_first_run.transactions.len(), 3);
+
+    let declare_tx_entry = tx_entries_after_first_run
+        .get("2341f038132e07bd9fa3cabf5fa0c3fde26b0fc03e7b09198dbd230e1b1e071c")
+        .unwrap();
+    assert_tx_entry_failed(declare_tx_entry, "declare", ScriptTransactionStatus::Error, vec!["Failed to find Not_this_time artifact in starknet_artifacts.json file. Please make sure you have specified correct package using `--package` flag and that you have enabled sierra and casm code generation in Scarb.toml."]);
+
+    let deploy_tx_entry = tx_entries_after_first_run
+        .get("2402e1bcaf641961a4e97b76cad1e91f9522e4a34e57b5f740f3ea529b853c8f")
+        .unwrap();
+    assert_tx_entry_failed(
+        deploy_tx_entry,
+        "deploy",
+        ScriptTransactionStatus::Fail,
+        vec!["Class with hash ClassHash", "is not declared"],
+    );
+
+    let invoke_tx_entry = tx_entries_after_first_run
+        .get("9e0f8008202594e57674569610b5cd22079802b0929f570dfe118b107cb24221")
+        .unwrap();
+    assert_tx_entry_failed(
+        invoke_tx_entry,
+        "invoke",
+        ScriptTransactionStatus::Fail,
+        vec!["Requested contract address", "is not deployed"],
+    );
+}
+
+#[tokio::test]
+async fn test_state_file_rerun_failed_tx() {
+    let script_dir = copy_script_directory_to_tempdir(
+        SCRIPTS_DIR.to_owned() + "/state_file/",
+        Vec::<String>::new(),
+    );
+
+    let accounts_json_path = get_accounts_path(ACCOUNT_FILE_PATH);
+
+    let script_name = "rerun_failed_tx";
+    let args = vec![
+        "--accounts-file",
+        accounts_json_path.as_str(),
+        "--account",
+        "user4",
+        "--url",
+        URL,
+        "script",
+        "run",
+        &script_name,
+    ];
+
+    let snapbox = runner(&args).current_dir(script_dir.path());
+
+    snapbox.assert().success().stdout_matches(indoc! {r"
+        ...
+        command: script run
+        status: success
+    "});
+
+    let state_file_path = Utf8PathBuf::from_path_buf(
+        script_dir
+            .path()
+            .join(get_default_state_file_name(script_name, "alpha-goerli")),
+    )
+    .unwrap();
+    let tx_entries_after_first_run = read_txs_from_state_file(&state_file_path).unwrap().unwrap();
```

Comment:
Let's read the state file before executing the script and use `assert_tx_entry_failed` to explicitly show that the state file has been overwritten.

Reply:
Nothing should be overwritten, the point is to rerun a script that should have an invoke call that fails, but the script succeeds (from what I understand)

Reply:
So what's the point of adding the file [rerun_failed_tx_alpha-goerli_state.json](https://github.com/foundry-rs/starknet-foundry/pull/1844/files#diff-8f17219dce3b6579af66d2d4c1c83524e209704104f172ac44e1d9e3798c1808)?

Reply:
Not sure tbh, my bet is that the intention was to make sure that failed tx do not overwrite state file when the status did not change, that's how I interpret this test tbh

Reply:
I understand it differently. The name contains `rerun_failed`, so I expect that this test was supposed to simulate that the previously failed transaction will be repeated and the state file will be updated upon success. In my opinion such a test makes a lot of sense.